src/org/opensolaris/opengrok/util/OptionParser.java

```diff
@@ -0,0 +1,701 @@
+                    String[] parts = argument.split("[ =]");
+                    
+                    if (parts.length == 1) {
+                        opt.addAlias(parts[0]);
+                    } else {
+                        opt.set(parts[0], parts[1]);
+                    }
+                } else if (argument.startsWith("-")) {
+                    // handle -x -xARG -x=ARG -x[OPT] -x[=OPT] -x PLACE
+                    String optName = argument.substring(0,2);
+                    String remainder = argument.substring(2);
+                    opt.set(optName, remainder);
+                    
+                } else if (argument.startsWith("=")) {
+                    opt.setArgument(argument.substring(1));
+                } else if (argument.startsWith("/")) {
+                    // regular expression (sans '/'s)
+                    opt.setPattern(argument.substring(1,argument.length()-1));
+                } else {
+                    // this is description
+                    opt.addDescription(argument);
+                }
+            // This is indicator for a set of specific allowable option values
+            } else if (arg instanceof String[]) {
+                opt.setAllowedValues((String[])arg);
+            // This is indicator for option value data type 
+            // to which the parser will take and convert.
+            } else if (arg instanceof Class) {
+                opt.setValueType((Class<?>)arg);
+            }
+        }
+        optionList.add(opt);
+        usageSummary.add(opt);
+        return opt;
+    }
+    
+    private String argValue(String arg, Boolean mandatory) {
+        // Initially assume that the given argument is going
+        // to be the option's value. Note that if the argument
+        // is actually another option (starts with '-') then
+        // there is no value available. If the option is required
+        // to have a value, null is returned. If the option
+        // does not require a value, an empty string is returned.
+        String value = arg;
+        Boolean isOption = value.startsWith("-");
+        
+        if (mandatory) {
+            if(isOption ) {
+                value = null;
+            }
+        } else if (isOption) {
+            value = "";
+        }
+        return value;
+    }
+    
+    private String getOption(String arg) {
+        String option = null;
+        
+        if (arg.startsWith("-")) {
+            if (arg.startsWith("--")) {
+                option = arg;                 // long name option (--longOption)
+            } else if (arg.length() > 2) {
+                option = arg.substring(0,2);  // short name option (-xValue)
+            } else {
+                option = arg;                 // short name option (-x)
+            }
+        }
+        return option;
+    }
+
+    // Allows user to enter initial substring of a long option name.
+    private boolean unknown(String option, int index) throws ParseException {
```

Comment:
this should be covered by a test

Reply:
how does one put a private method in a test?

Reply:
Use protected